File: backend/src/main/scala/microservices/operations/airline/objects/ManagerCabinInventoryPlannerResponse.scala
// 这个文件定义 airline 航班舱位库存响应。
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ManagerCabinInventoryPlannerResponse(
    inventoryId: String,
    cabinClass: String,
    availableSeats: Int,
    unitPrice: String,
    currency: String,
    status: String,
    isBookable: Boolean
)
object ManagerCabinInventoryPlannerResponse:
  given sourceEncoder: Encoder[ManagerCabinInventoryPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[ManagerCabinInventoryPlannerResponse] = deriveDecoder

File: frontend/src/microservices/operations/airline/objects/ManagerCabinInventoryPlannerResponse.ts
// 本文件定义航空管理端的舱位库存响应结构。

export type ManagerCabinInventoryPlannerResponse = {
  inventoryId: string
  cabinClass: string
  availableSeats: number
  unitPrice: string
  currency: string
  status: string
  isBookable: boolean
}

export const managerCabinInventoryPlannerResponseFromJson = (json: string): ManagerCabinInventoryPlannerResponse =>
  JSON.parse(json) as ManagerCabinInventoryPlannerResponse

export const managerCabinInventoryPlannerResponseToJson = (value: ManagerCabinInventoryPlannerResponse): string =>
  JSON.stringify(value)
